src/coreclr/jit/emitxarch.cpp

```diff
@@ -6811,6 +6826,12 @@ void emitter::emitIns_R_R_C(
     id->idReg1(reg1);
     id->idReg2(reg2);
     id->idAddr()->iiaFieldHnd = fldHnd;
+#if defined(TARGET_XARCH)
+    if ((instOptions == INS_OPTS_EVEX_b) && UseEvexEncoding())
+    {
+        id->idSetEvexbContext();
+    }
```

Comment:
We should never get here or have the option set if EVEX isn't supported:
```suggestion
    if (instOptions == INS_OPTS_EVEX_b)
    {
        assert(UseEvexEncoding());
        id->idSetEvexbContext();
    }
```

If we did get here, otherwise, then we probably have done an incorrect optimization higher up.

Reply:
Same elsewhere in `emitxarch.cpp`